Client/InfluxDB.Client.Api/Client/ApiClient.cs

```diff
@@ -115,9 +116,13 @@ public ApiClient(string basePath = "http://localhost/api/v2")
             string path, Method method, List<KeyValuePair<string, string>> queryParams, object postBody,
             Dictionary<string, string> headerParams, Dictionary<string, string> formParams,
             Dictionary<string, FileParameter> fileParams, Dictionary<string, string> pathParams,
-            string contentType)
+            string contentType,
+            HttpCompletionOption httpCompletionOption = HttpCompletionOption.ResponseContentRead)
```

Comment:
@bednar I am still hitting the buffer limit in 4.16.0 for large queries was the default suppose to be  HttpCompletionOption.ResponseHeadersRead?

Reply:
This is the default value that should be overridden in our files. It seems like `HttpCompletionOption.ResponseHeadersRead` should also be set in the `QueryApi` file. As a workaround, can you try setting `HttpClient.MaxResponseContentBufferSize` in `HttpClient`? How to use a custom `HttpClient` is mentioned here: https://github.com/influxdata/influxdb-client-csharp/pull/528.

For more information, see:

- https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.maxresponsecontentbuffersize?view=net-8.0

Reply:
I've created a prerelease version of the client with a fix from #649. Can you test the version `4.17.0-dev.headers.read.1`? You can find it here: https://www.nuget.org/packages/InfluxDB.Client/4.17.0-dev.headers.read.1.

Reply:
Hi thank you for the response. I tested the change for https://github.com/influxdata/influxdb-client-csharp/pull/649 and it fixed the issue.